File: backend/SitPass/src/main/java/com/example/SitPass/model/Rate.java
package com.example.SitPass.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Entity
@Table(name = "rate")
public class Rate {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Column
    private Integer equipment;

    @Column
    private Integer staff;

    @Column
    private Integer hygene;

    @Column
    private Integer space;

    @Column
    private Double average;

    @ManyToOne
    @JoinColumn(name = "review_id")
    @JsonIgnore
    private Review review;



}

File: frontend/SitPass/src/app/models/Rate.ts
// rate.model.ts
import { Review } from './Review'; // Uveri se da imaš odgovarajući Review model

export interface Rate {
  id?: number;          // `Long` u Javi se preslikava na `number` u TypeScript-u
  equipment?: number;   // `Integer` u Javi se preslikava na `number` u TypeScript-u
  staff?: number;       // Isto kao gore
  hygene?: number;      // Isto kao gore
  space?: number;       // Isto kao gore
  review?: Review;      // Preslikava `Review` objekat
  average?: number;     // `Double` u Javi se preslikava na `number` u TypeScript-u
}
